Common/src/toolboxes/C1DInterpolation.cpp

```diff
@@ -0,0 +1,202 @@
+/*!
+ * \file inlet_interpolation_functions.cpp
+ * \brief Inlet_interpolation_functions
+ * \author Aman Baig
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation 
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+
+#include "../../include/toolboxes/C1DInterpolation.hpp"
+
+/*--- Function for evaluating the value for akima spline ---*/
+su2double CAkimaInterpolation::EvaluateSpline(su2double Point_Interp){
+    Point_Match = true;
+    int i =0, j=this->n-1;
+
+    while (j-i>1){ 
+        int m=(i+j) / 2 ; 
+            if (Point_Interp>this->x[m]) {i=m;}
+            else {j=m;} 
+    }
+
+    su2double h=Point_Interp-this->x[i] ;
+    return this->y[i]+h*(this->b[i]+h*(this->c[i]+h*this->d[i])) ;
+}
+
+/*--- Function for evaluating the value for linear 'spline' ---*/
+su2double CLinearInterpolation::EvaluateSpline(su2double Point_Interp){
+    int size = this->x.size();
+
+    for (int i=0;i<size-1;i++){
+        if(Point_Interp>=this->x[i] && Point_Interp<=this->x[i+1]){
+            this->Point_Match = true;
+            return (Point_Interp-this->x[i])*dydx[i]+this->y[i];}
+    }
+    return 0;
+}
+
+/*--- Function for setting the cofficients for linear 'spline' ---*/
+void CLinearInterpolation::SetSpline(vector<su2double> &X, vector<su2double> &Data){
+    int n = X.size();
+    vector<su2double> h (n-1);
+    this->x.resize(n);
+    this->y.resize(n);
+    this->dydx.resize(n-1);
+
+    for (int i=0; i<n ; i++){
+        this->x[i] = X[i];
+        this->y[i] = Data[i];
+    }
+
+    for (int i=0; i<n-1;i ++){h[i]=X[i+1]-X[i];}
+    for (int i=0; i<n-1;i++) {this->dydx[i]=(Data[i+1]-Data[i]) / h [i];}
```

Comment:
`h` does not need to be a vector, you can combine the 2 last loops into just one.

Reply:
I agree.